test/test_automl/automl_utils.py

```diff
@@ -55,3 +56,74 @@ def count_succeses(cv_results):
         [status in ['Success', 'Success (but do not advance to higher budget)']
          for status in cv_results['status']]
     )
+
+
+class AutoMLLogParser(object):
+    def __init__(self, logfile):
+        self.logfile = logfile
+        self.lines = self.parse_logfile()
+
+    def parse_logfile(self):
+        # We care about the [debug/info/...] messages
+        assert os.path.exists(self.logfile), "{} not found".format(self.logfile)
+
+        with open(self.logfile) as f:
+            content = f.readlines()
+        content = [x.strip() for x in content if re.search(r'[\w+]', x)]
+        return content
+
+    def count_ensembler_iterations(self):
+        # we have to count the start msg from pynisher
+        # and the return msg
+        # We expect the start msg to be something like:
+        # [DEBUG] [2020-11-26 19:22:42,160:EnsembleBuilder] Function called with argument: (61.
+        iterations = []
+
+        # One thing is to call phynisher, the other is to actually execute the funciton
+        iterations_from_inside_ensemble_builder = []
+        for line in self.lines:
+
+            # Pynisher call
+            match = re.search(
+                r'EnsembleBuilder]\s+Function called with argument:\s+\(\d+\.\d+, (\d+), \w+',
```

Comment:
Don't you have to escape the `]`?

Reply:
actually, if we escape it will fail because I am looking for ]. If you escape it, it assumes it is the bracket regexp that says any-of